addons/mail/static/src/messaging/component/discuss_sidebar/discuss_sidebar.js

```diff
@@ -17,51 +17,76 @@ class DiscussSidebar extends Component {
      */
     constructor(...args) {
         super(...args);
-        this.state = useState({ quickSearchValue: "" });
-        this.storeGetters = useGetters();
-        this.storeProps = useStore((...args) => this._useStoreSelector(...args), {
-            compareDepth: () => this._useStoreCompareDepth(),
-        });
+        useStore(
+            (...args) => this._useStoreSelector(...args),
+            { compareDepth: () => this._useStoreCompareDepth() }
+        );
+
         /**
          * Reference of the quick search input. Useful to filter channels and
          * chats based on this input content.
          */
-        this._quickSearchRef = useRef('quickSearch');
+        this._quickSearchInputRef = useRef('quickSearchInput');
+
+        // bind since passed as props
```

Comment:
Isn't that bad design? Shouldn't we use trigger and/or store?

Reply:
Don't think so, bind is just to prevent useless re-render of child when parent is re-rendered. It makes sense to pass function for generic components like `AutocompleteInput`